components/formats-gpl/src/loci/formats/in/MetamorphReader.java

```diff
@@ -871,12 +874,13 @@ else if (firstSeriesChannels != null) {
 
           if ((int) wave[waveIndex] >= 1) {
             // link LightSource to Image
+            int laserIndex = i * getEffectiveSizeC() + c;
```

Comment:
As there is now only instrument/detector per fileset, should the DetectorSettings object setters above be updated in a similar manner by creating a `detectorSettingsIndex` using the series/channel and calling `setDetectorSettingsxxx(value, 0, settingsIndex)`?